src/main/java/org/wildfly/security/auth/provider/ldap/UserPasswordCredentialLoader.java

```diff
@@ -71,43 +53,21 @@
 
     static final String DEFAULT_USER_PASSWORD_ATTRIBUTE_NAME = "userPassword";
 
-    public static final Set<String> SUPPORTED_ALGORITHMS = Collections.unmodifiableSet(
-        new HashSet<>(
-            Arrays.asList(ALGORITHM_CLEAR, ALGORITHM_SIMPLE_DIGEST_MD5, ALGORITHM_SIMPLE_DIGEST_SHA_1,
-                ALGORITHM_SIMPLE_DIGEST_SHA_256, ALGORITHM_SIMPLE_DIGEST_SHA_384, ALGORITHM_SIMPLE_DIGEST_SHA_512,
-                ALGORITHM_PASSWORD_SALT_DIGEST_MD5, ALGORITHM_PASSWORD_SALT_DIGEST_SHA_1, ALGORITHM_PASSWORD_SALT_DIGEST_SHA_256,
-                ALGORITHM_PASSWORD_SALT_DIGEST_SHA_384, ALGORITHM_PASSWORD_SALT_DIGEST_SHA_512, ALGORITHM_BSD_CRYPT_DES, ALGORITHM_CRYPT_DES)
-            )
-        );
-
     private final String userPasswordAttributeName;
 
-    // TODO - withCertaintly could be stores with the values in the map so some credential names are always supported and some only possible.
-    // Also control of which credentials can be persisted could have similar treatment.
-
-    private final boolean withCertainty;
-    private final Map<String, Set<String>> credentialNameAlgorithms;
-
-    UserPasswordCredentialLoader(final boolean withCertainty, final Map<String, Set<String>> credentialNameAlgorithms) {
```

Comment:
Just wondering if the simplification may be a step too far, it feels as though an administrator may still want to be able to fine tune which algorithms are supported here.  The withCertainty parameter also gave the ability to indicate of all users are represented equally.

Later on when we are back at configuring mechanism availability I think there could be a case to control mechanisms offered based on which ones are definitely supported compared to the ones that might be supported.